src/main/java/seedu/algobase/logic/parser/ParserUtil.java

```diff
@@ -293,47 +291,66 @@ public static LocalDate parseDate(String date) throws ParseException {
      *
      * @throws ParseException if the given {@code string tabType} is invalid.
      */
-    public static TabType parseTabType(String tabType) throws ParseException {
+    public static TabType parseTabType(String tabType, String errorMessage) throws ParseException {
+        // Check if valid string
         try {
             return TabType.valueOf(tabType.toUpperCase());
-        } catch (IllegalArgumentException e) {
-            throw new ParseException(e.toString());
+        } catch (IllegalArgumentException ire) {
+            // Try if the index is valid
+        }
+
+        // Check if valid index
+        try {
+            Index tabTypeIndex = Index.fromOneBased(Integer.parseInt(tabType));
+            return TabType.values()[tabTypeIndex.getZeroBased()];
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+            throw new ParseException(errorMessage);
         }
```

Comment:
It would be more neat if you can write `checkValidTabTypeString` and `checkValidTabTypeIndex` to avoid empty catch block.

Reply:
What do you mean?